modules/rosapps/applications/sysutils/man/man.c

```diff
@@ -88,33 +88,40 @@ Usage()
 int
 AnalyzeArgv(char *argument)
 {
-    int element=0;
+    int element;
     char HelpFlag=0;
     char *keys[]={"--help","/h","/?","-h"};
     char *sections[]={".1",".2",".3",".4",".5",".6",".7",".8",".9"};
     char *filename=(char*)malloc(sizeof(char)*MAXLINE);
 
     strcpy(filename,argument); //save argument value
 
-    for(element=0; element < 5;element++)
+    for(element = 0; element < 4; element++)
+    {
      if(!strcmp(keys[element],argument))
      {
       Usage();
       HelpFlag=1;
      }
+    }
 
    element = 0;
 
    if(!HelpFlag)
+   {
 
    if(OpenF(filename))
-    while(OpenF(strcat(filename,sections[element])) && (element<9))
+   {
+    while(element < 9 && OpenF(strcat(filename, sections[element])))
     {
      strcpy(filename,argument);
      element++;
     }
 
     if(element>8) printf("No manual for %s\n",argument);
```

Comment:
And here it would be something like `if (element >= _countof(sections))`.

Reply:
Switched to `_countof()`.